compiler/test/dotty/tools/vulpix/ParallelTesting.scala

```diff
@@ -1088,7 +1088,10 @@ trait ParallelTesting extends RunnerOrchestration { self =>
     targetDir.mkdirs()
     assert(targetDir.exists, s"couldn't create target directory: $targetDir")
 
-    val target = JointCompilationSource(s"$testName from $callingMethod", files.map(new JFile(_)).toArray, flags, targetDir)
+    // Sort the files so that the compilation order is deterministic.
+    // See https://github.com/lampepfl/dotty/issues/2792
+    val sorted = files.map(new JFile(_)).toArray.sorted
```

Comment:
https://github.com/lampepfl/dotty/pull/2193/files already has some code to sort things with some specific seed, it would be nice to do some refactoring so that it is used here too.